### PR TITLE
Upgrade RevApi Dependency

### LINKED ISSUE #22692: Upgrade RevApi and RevApi Plugin Versions
Currently, RevApi `0.20.0` and RevApi Maven plugin `0.11.2` are being used, these versions are roughly 18 months older. These dependencies should be upgraded to later versions of `0.24.3` and `0.14.3`.

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Fixes #22692

This PR upgrades the versions of RevApi being used to their latest releases. Migrating to the new RevApi version required changes to how RevApi is configured, so due to that this PR additionally includes a clean-up of RevApi configurations in each POM file that was previously configuring RevApi. A new configuration property, `revapi.skip`, was added to the parent POM to simplify configuring skipping RevApi analysis and the RevApi ignore configurations that were repeated in resourcemanager projects were added into the central ignore configurations.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 1]
```diff
@@ -2,426 +2,458 @@
...
+          "class com\\.microsoft\\.azure\\..+",
+          "class javax\\.jms(\\..+)?",
+          "class javax\\.servlet(\\..+)?",
+          "class io\\.micrometer(\\..+)?",
+          "class com\\.azure\\.data\\.cosmos(\\..+)?",
+          "class com\\.azure\\.spring\\.data\\.cosmos\\..+",
+          "class com\\.azure\\.spring\\.data\\.gremlin\\..+",
+          "class com\\.azure\\.spring(\\..+)?",
+          "class org\\.apache\\.avro\\..+",
+          "class com\\.azure\\.resourcemanager\\..*\\.fluent\\..*"
```
Comment (srnagar) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:68]: Not all resourcemanager libraries exclude this package. For example, [applicationinsights](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/applicationinsights/azure-resourcemanager-applicationinsights/pom.xml) doesn't skip this.

File: sdk/cosmos/azure-cosmos-encryption/pom.xml [hunk 2]
```diff
@@ -37,6 +37,10 @@ Licensed under the MIT License.
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <jacoco.min.linecoverage>0.01</jacoco.min.linecoverage>
     <jacoco.min.branchcoverage>0.01</jacoco.min.branchcoverage>
+
+    <!-- CosmosSkip - This is not a module we want/expect external customers to consume. Skip breaking API checks. -->
+    <!-- This can only be enabled once we release GA, as it needs a stable version to check for breaking changes. -->
+    <revapi.skip>true</revapi.skip>
```
Comment (srnagar) [inline sdk/cosmos/azure-cosmos-encryption/pom.xml:43]: azure-cosmos-encryption is going to be customer-facing and it shouldn't skip revapi checks. This library is still in preview and skipping revapi is not required anyway I guess.

cc: @kushagraThapar

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/pom.xml [hunk 3]
```diff
@@ -29,6 +29,7 @@
     <legal>
       <![CDATA[[INFO] Any downloads listed may be third party software.  Microsoft grants you no rights for third party software.]]>
     </legal>
+    <revapi.skip>true</revapi.skip>
```
Comment (srnagar) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/pom.xml:32]: This is a GA library used by customers and shouldn't probably skip revapi checks. 

cc: @kushagraThapar

File: sdk/cosmos/azure-cosmos/pom.xml [hunk 4]
```diff
@@ -53,6 +53,10 @@ Licensed under the MIT License.
       --add-opens com.azure.cosmos/com.azure.cosmos.implementation.throughputControl.controller.request=ALL-UNNAMED
       --add-opens com.azure.cosmos/com.azure.cosmos.implementation.clienttelemetry=ALL-UNNAMED
     </javaModulesSurefireArgLine>
+
+    <!-- CosmosSkip - This is not a module we want/expect external customers to consume. Skip breaking API checks. -->
```
Comment (srnagar) [inline sdk/cosmos/azure-cosmos/pom.xml:57]: azure-cosmos is for external customers and shouldn't skip revapi checks.

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json (hunk 1 above)
Comment (alzimmermsft) [reply eng/code-quality-reports/src/main/resources/revapi/revapi.json:68]: I wonder if that is due to the fact it has never GA'd?

**Pull request closed without merging**